src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -57,11 +58,21 @@
     private final ImportHandler importHandler;
     private final CustomLocalDragboard localDragboard;
 
+    private long lastKeyPressTime;
+    private String columnSearchTerm;
+
     public MainTable(MainTableDataModel model, JabRefFrame frame,
                      BasePanel panel, BibDatabaseContext database,
                      MainTablePreferences preferences, ExternalFileTypes externalFileTypes, KeyBindingRepository keyBindingRepository) {
         super();
 
+        this.setOnKeyTyped(key -> {
+            if (this.getSortOrder().size() == 0) {
```

Comment:
Please check for emptyness using `.isEmpty()`.

Reply:
Yeah, It is much better than checking the size, I will modify it. Thanks for your suggestions